### DISCUSSION TITLE
Instance Bots

### DISCUSSION THREAD

**Discussion by RomashinTimofey** [trickerer/Trinity-Bots#724] [Ideas]
The suggestion is that any dungeon should have a portal at the beginning that has a Menu. The menu has a choice of whom you want to hire, tank, healer, rdd, mdd. After the summoning, these bots live and obey the player only inside this dungeon, for the duration of its passage, and are deleted and unlinked after its completion or if the player has left this dungeon.

**Comment by trickerer** [comment]
Any kind of short-term owned bots is a concept I abandoned long ago. I'm not doing gear generation for player owned bots, gearing bots is player's job and they're not very useful naked.

**Comment by RomashinTimofey** [comment]
I understand you. But could you at least suggest, having such a goal, how it could be implemented, and which classes will be involved for this purpose? I want to be able to hire bots only for the duration of the dungeon, I am grateful in advance for the answer.

**Reply by trickerer** [reply]
You'd need bot (+ gear) generation mechanism similar to BG wandering bots but player owned instead and deletion mechanism at player's teleport from instance map. New class - InstanceBotGenerator, maybe 1 or 2 Events to schedule, again, similar to BG bots. Bot roles is a problem since player has to set them every time. A number of bot commands will be a problem since they expect bot name / class. A new creature with its own gossip script (separate file and a new class), similar to botgiver.

**Reply by RomashinTimofey** [reply]
I just wanted to thank you for such a detailed answer. Unfortunately, I'm not good enough in C++ to figure this out, so the above looks overwhelming ^^'
